### PR TITLE
Update eclipse to version 4.11.0 (fix #381)

### LINKED ISSUE #381: Add support for Eclipse 4.11.0 auto formatter
Eclipse 4.11 (2019/3) was released recently so it would be nice to update the support for it.

### PR DESCRIPTION
**Pull request by iemejia**

### CODE REVIEW

File: lib-extra/src/main/resources/com/diffplug/spotless/extra/eclipse_jdt_formatter/v4.11.0.lockfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Spotless formatter based on JDT version 4.11.0 (see https://projects.eclipse.org/projects/eclipse.jdt)
+# Compare tag in M2 pom with https://git.eclipse.org/c/jdt/eclipse.jdt.core.git/log/?h=R4_9_maintenance to determine core version.
```
Comment (fvgh) [inline lib-extra/src/main/resources/com/diffplug/spotless/extra/eclipse_jdt_formatter/v4.11.0.lockfile]: It should state https://git.eclipse.org/c/jdt/eclipse.jdt.core.git/tag/?h=R4_11 now

File: lib-extra/src/test/java/com/diffplug/spotless/extra/java/EclipseJdtFormatterStepTest.java [hunk 2]
```diff
@@ -24,7 +24,8 @@
 
 	@Override
 	protected String[] getSupportedVersions() {
-		return new String[]{"4.6.1", "4.6.2", "4.6.3", "4.7.0", "4.7.1", "4.7.2", "4.7.3a", "4.8.0", "4.9.0", "4.10.0"};
+		return new String[]{"4.6.1", "4.6.2", "4.6.3", "4.7.0", "4.7.1", "4.7.2", "4.7.3a", "4.8.0", "4.9.0", "4.10.0",
```
Comment (fvgh) [inline lib-extra/src/test/java/com/diffplug/spotless/extra/java/EclipseJdtFormatterStepTest.java]: The formatting is broken, that's why the build fails.
Run 'gradlew spotlessApply' to fix these violations.

File: lib-extra/src/main/resources/com/diffplug/spotless/extra/eclipse_jdt_formatter/v4.11.0.lockfile (hunk 1 above)
Comment (iemejia) [reply lib-extra/src/main/resources/com/diffplug/spotless/extra/eclipse_jdt_formatter/v4.11.0.lockfile]: Done and updated to the four latest revisions

File: lib-extra/src/test/java/com/diffplug/spotless/extra/java/EclipseJdtFormatterStepTest.java (hunk 2 above)
Comment (iemejia) [reply lib-extra/src/test/java/com/diffplug/spotless/extra/java/EclipseJdtFormatterStepTest.java]: Done, it catched a .gradle formatting error too so I included it in the PR

**Comment by iemejia** [comment]
Rebased and fixed issues. @fvgh PTAL

File: lib-extra/src/test/java/com/diffplug/spotless/extra/java/EclipseJdtFormatterStepTest.java (hunk 2 above)
Comment (fvgh) [reply lib-extra/src/test/java/com/diffplug/spotless/extra/java/EclipseJdtFormatterStepTest.java]: :+1:

**Comment by nedtwigg** [comment]
Published in x.21.0.  Thanks for the code PR @iemejia.  Let me know if you'd like any changes [to this credit](https://github.com/diffplug/spotless/commit/b4378254edacd9ec30d675e72baf4b8abfc7d460).

**Comment by iemejia** [comment]
That's perfect, thanks!

**Pull request merged**